CONTRIBUTING.md

```diff
@@ -1,19 +1,114 @@
-# Contributing
+# Contributing guidelines
 
 When contributing to this repository, please first discuss the change you wish to make via issue,
-email, or any other method with the owners of this repository before making a change. 
+email, or any other method with the owners of this repository before making a change.
 
 Please note we have a code of conduct, please follow it in all your interactions with the project.
 
+### Setting things up
+
+To set up the development environment, follow the instructions in README.
+
+### Finding something to work on
+
+The issue tracker of EvalAI a good place to start. If you find something that interests you, comment on the thread and we’ll help get you started.
+
+Alternatively, if you come across a new bug on the site, please file a new issue and comment if you would like to be assigned. The existing issues are tagged with one or more labels, based on the part of the website it touches, its importance etc., that can help you in selecting one.
+
+If neither of these seem appealing, please post on our channel and we will help find you something else to work on.
+
+### Instructions to submit code
+
+Before you submit code, please talk to us via the issue tracker so we know you are working on it.
+
+Our central development branch is development. Coding is done on feature branches based off of development and merged into it once stable and reviewed. To submit code, follow these steps:
```

Comment:
I noticed your primary branch is not `development`.

Reply:
Changed this as well. Please check now.